sched/mqueue/mq_timedreceive.c

```diff
@@ -134,9 +127,11 @@ static void nxmq_rcvtimeout(wdparm_t pid)
  *
  ****************************************************************************/
 
-ssize_t file_mq_timedreceive(FAR struct file *mq, FAR char *msg,
-                             size_t msglen, FAR unsigned int *prio,
-                             FAR const struct timespec *abstime)
+static ssize_t
+file_mq_timedreceive_internal(FAR struct file *mq, FAR char *msg,
+                              size_t msglen, FAR unsigned int *prio,
+                              FAR const struct timespec *abstime,
+                              sclock_t ticks, bool isabs)
```

Comment:
why not check abstime != NULL directly, but let's caller pass isabs?

Reply:
abstime and tickets may both be passed by the caller,  sometimes they may be invalid values

Reply:
caller can check abstime before calling file_mq_timedreceive_internal

Reply:
@anchao how about this?

Reply:
ok, it seems that the isabs check could be removed. I tested it in Linux and found that if the timeout is NULL, the buffer can be received:
`ret = mq_timedreceive(&mq, buffer, msgsize, &prio, NULL);`